### DISCUSSION TITLE
Is there any linter can check missing a blank line between functions?

### DISCUSSION THREAD

**Discussion by ppd0705** [golangci/golangci-lint#2946] [Q&A] [answered]
I want keep a code format: add a blank line between functions. 
It seems like `gofmt` does not support this  ( related issue: https://github.com/golang/go/issues/22337)

**Comment by bombsimon** [comment, marked as answer]
[`gofumpt`](https://github.com/mvdan/gofumpt) has the rule _"Multiline top-level declarations must be separated by empty lines"_ which affects functions at top level scope. It will change

```go
func foo() {
	println("multiline foo")
}
func bar() {
	println("multiline bar")
}
```

To

```go
func foo() {
	println("multiline foo")
}

func bar() {
	println("multiline bar")
}
```

[`wsl`](https://github.com/bombsimon/wsl) has the rule [_"Expressions should not be cuddled with blocks"_](https://github.com/bombsimon/wsl/blob/master/doc/rules.md#expressions-should-not-be-cuddled-with-blocks) although WSL does not have an auto fixer like `gofumpt` and it's also not possible to only enable this check in WSL.

**Reply by ppd0705** [reply]
@bombsimon  thank you very much, [gofumpt](https://github.com/mvdan/gofumpt) meets my need.
